packages/edit-site/src/components/page-templates/index.js

```diff
@@ -92,11 +93,22 @@ export default function PageTemplates() {
 		};
 	}, [ layout, activeView ] );
 	const [ view, setView ] = useState( defaultView );
+
+	// Sync the layout from the URL to the view state.
+	const onChangeUrlLayout = useEvent( () => {
+		setView( ( currentView ) => ( {
+			...currentView,
+			type: layout ?? DEFAULT_VIEW.type,
+		} ) );
+	} );
 	useEffect( () => {
-		const usedType = layout ?? DEFAULT_VIEW.type;
+		onChangeUrlLayout();
```

Comment:
Do we need this effect?

Reply:
What do you mean? If the "layout" in the url changes, we need to reflect that in the UI

Reply:
Isn't this already handled by `onChangeView` and on `defaultView` when we load the page? By commenting this code, my results in testing were identical..

Could this be related to?
>For me in trunk templates seem to work 🤔

Reply:
In trunk, "search" is preserved but not "filters", in this PR both are preserved

Reply:
- onChangeView: this is the opposite thing, if a user changes the layout in the UI, it gets reflected in the url
 - defaultView: this is about the initial state (when the component is mounted but it's not used when the component is re-rendered, for instance when you change the url manually)

Reply:
>In trunk, "search" is preserved but not "filters", in this PR both are preserved

I was testing sort and search.. You're right.

Reply:
>onChangeView: this is the opposite thing, if a user changes the layout in the UI, it gets reflected in the url
defaultView: this is about the initial state (when the component is mounted but it's not used when the component is re-rendered, for instance when you change the url manually)

I know, but I couldn't trigger a rerender when I changed the url manually. Am I missing something?

Can you elaborate a bit on testing instructions to see when we use this effect except for the initial rendering?

Reply:
>I know, but I couldn't trigger a rerender when I changed the url manually. Am I missing something?

Maybe it's not possible right now in the UI but the url is not a state controlled by this component, someone can trigger `navigate` elsewhere changing the layout, so the component should know how to adapt to this properly.

Reply:
Ok then. I just thought if someone would want to change the layout, would do it with `changeView`.

Reply:
Who "someone" is, is the important bit here. We're modifying a component, so that "someone" is not necessarily the "user", it can be any other developer within our outside Gutenberg. 

A developer calling a `navigate` shouldn't have to go to a random component to ensure that component reacts to the url change properly when that url is not something new in the app.

This is more about code quality of separation of concern. If a state is managed externally to a component and that state can be changed (it's not immutable), then the component needs to implement logic to react to that state change.